data/dbHelpers.js

```diff
@@ -0,0 +1,62 @@
+const knex =require("knex")
+const dbConfig = require("../knexfile.js"); 
+const db = knex(dbConfig.development);
+
+module.exports ={
+
+    getDishes:()=>{
+        return db('dishes')
+
+    },
+    getById : (id)=> {
+        return db('dishes')
+          .where({ id })
+          .first();
+    },
+    addDish: (dish) => {
+        return db('dishes')
+          .insert(dish)
+          .then(id => {
+            console.log("IDS",id)
+            return id[0];
+          });
+    },
+    addRecipe:(recipe)=>{
+        return db('recipes')
+        .insert(recipe)
+        .then(id=>{//aaray of recipes id where only one element is there in the array
+          return id[0];//it givers the  id of the inserted recipe, 
```

Comment:
Another way of getting the first element off an array here could be the following:
```javascript
.then(([ id ]) => {
  return id;
});
```
That way uses [array destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring). We could simplify it further by using an [implicit return](https://jaketrent.com/post/javascript-arrow-function-return-rules/):
```javascript
.then(([ id ]) => id);
```